db/changes.go

```diff
@@ -388,10 +391,17 @@ func (db *Database) SimpleMultiChangesFeed(chans base.Set, options ChangesOption
 				// Ensure backfill isn't already in progress for this seqAddedAt
 				backfillPending := options.Since.TriggeredBy == 0 || options.Since.TriggeredBy < seqAddedAt
 
+				backfillInOtherChannel := options.Since.TriggeredBy != 0 && options.Since.TriggeredBy > seqAddedAt
+
+
 				if isNewChannel || (backfillRequired && backfillPending) {
 					// Newly added channel so initiate backfill:
 					chanOpts.Since = SequenceID{Seq: 0, TriggeredBy: seqAddedAt}
+				} else if (backfillInOtherChannel){
+					chanOpts.Since = SequenceID{Seq: options.Since.TriggeredBy, TriggeredBy: seqAddedAt}
```

Comment:
I don't think TriggeredBy should be set in the SequenceID in this case - we just want the Seq set.

Can you put together a unit test to validate the change, based on the functional test that's failing?